frontend/pages/reports.js

```diff
@@ -0,0 +1,262 @@
+
+  const [reports, setReports] = useState([])
+  const [reportFlag, updateReportFlag] = useState(false)
+
+  useEffect(() => {
+    doApiRequest('/reports/?format=json')
+      .then(resp => {
+        if (resp.ok) {
+          return resp.json()
+        } else {
+          return []
+        }
+      })
+      .then(data => setReports(data))
+  }, [reportFlag])
+
+  const [includedFields, setIncludedFields] = useState(
+    (() => {
+      const initial = {}
+      Object.keys(fields).forEach(group =>
+        fields[group].forEach(f => {
+          initial[f] = false
+        })
+      )
+      return initial
+    })()
+  )
+
+  const query = {
+    format: 'xlsx',
+    fields: Object.keys(includedFields)
+      .filter(field => includedFields[field])
+      .map(name => nameToCode[name])
+      .filter(e => e !== undefined),
+  }
+
+  const generateCheckboxGroup = (groupName, fields) => {
+    return (
+      <div key={groupName} style={{ flexBasis: '50%', flexShrink: 0 }}>
+        <GroupLabel
+          key={groupName}
+          className="subtitle is-4"
+          style={{ color: CLUBS_GREY }}
+        >
+          {groupName}
+        </GroupLabel>
+        {fields.map((field, idx) => (
+          <div key={idx}>
+            <Checkbox
+              id={field}
+              checked={includedFields[field]}
+              onChange={() => {
+                setIncludedFields(prev => ({ ...prev, [field]: !prev[field] }))
+              }}
+            />
+            {'  '}
+            <CheckboxLabel htmlFor={field}>{field}</CheckboxLabel>
+          </div>
+        ))}
+      </div>
+    )
+  }
+
+  return (
+    <Container>
+      <WideContainer>
+        <h1 className="title" style={{ color: CLUBS_GREY }}>
+          Run existing report
+        </h1>
+        <div className="box">
+          <table className="table" style={{ width: '100%' }}>
+            <thead>
+              <tr>
+                <th>Name</th>
+                <th>Description</th>
+                <th>Actions</th>
+              </tr>
+            </thead>
+            <tbody>
+              {reports.map((report, i) => (
+                <tr key={i}>
+                  <td>{report.name || <Empty>None</Empty>}</td>
+                  <td>{report.description || <Empty>None</Empty>}</td>
+                  <td>
+                    <div className="buttons">
+                      <a
+                        href={(() => {
+                          const params = JSON.parse(report.parameters)
+                          const paramString = Object.keys(params).reduce(
+                            (acc, cur) =>
+                              `${acc}&${encodeURIComponent(
+                                cur
+                              )}=${encodeURIComponent(params[cur])}`,
+                            ''
+                          )
+                          return `${API_BASE_URL}/clubs/?existing=true${paramString}`
+                        })()}
```

Comment:
would be good to split things like this into helper functions. I prefer render methods being exclusively rendering information + calling helper functions.